tools/Runtime Condenser/Main.cpp

```diff
@@ -155,261 +167,141 @@ bool readFromFile()
-
-			if(numRuntime[i] != 0) outputFile << endl << endl << "The following runtime has occured " << numRuntime[i] << " time(s).\n";
-			if(storedRuntime[i] != "Blank") outputFile << storedRuntime[i] << endl;
-			if(storedProc[i] != "Blank") outputFile << storedProc[i] << endl;
-			if(storedSource[i] != "Blank") outputFile << storedSource[i] << endl;
-			if(storedUsr[i] != "Blank") outputFile << storedUsr[i] << endl;
-			if(storedSrc[i] != "Blank") outputFile << storedSrc[i] << endl;
+		for (int i=0; i < storedRuntime.size(); i++) {
+			runtime* R = storedRuntime[i];
+			outputFile << endl << endl << "The following runtime has occurred " << R->count << " time(s).\n"; 
+			outputFile << R->text << endl;
+			if(R->proc.length()) 
+				outputFile << R->proc << endl;
+			if(R->source.length()) 
+				outputFile << R->source << endl;
+			if(R->usr.length()) 
+				outputFile << R->usr << endl;
+			if(R->src.length()) 
+				outputFile << R->src << endl;
+			if(R->loc.length()) 
+				outputFile << R->loc << endl;
 		}
+		outputFile << endl << endl; //For spacing
 		
 		//and finally, hard deletes
-		if(totalHardDels > 0)
-		{
+		if(totalHardDels > 0) {
 			outputFile << endl << "** Hard deletions **";
-			for(int i=0; i <= maxStorage; i++)
-			{
-				if(numHardDel[i] != 0 && storedHardDel[i] != "Blank") outputFile << endl << storedHardDel[i] << " - " << numHardDel[i] << " time(s).\n";
+			for(int i=0; i < storedHardDel.size(); i++) {
+				harddel* D = storedHardDel[i];
+				outputFile << endl << D->type << " - " << D->count << " time(s).\n";
 			}
 		}
 		outputFile.close();
-	}
-	else
-	{
+	} else {
 		return false;
 	}
 	return true;
 }
 
-void sortRuntimes()
-{
-	string tempRuntime[maxStorage+1];
-	string tempProc[maxStorage+1];
-	string tempSource[maxStorage+1];
-	string tempUsr[maxStorage+1];
-	string tempSrc[maxStorage+1];
-	unsigned int tempNumRuntime[maxStorage+1];
-	unsigned int highestCount = 0; //Used for descending order
-//	int keepLooping = 0;
-
-	//Move all of our data into temporary arrays. Also clear the stored data (not necessary but.. just incase)
-	for(int i=0; i <= maxStorage; i++)
-	{
-		//Get the largest occurance of a single runtime
-		if(highestCount < numRuntime[i])
-		{
-			highestCount = numRuntime[i];
-		}
 
-		tempRuntime[i] = storedRuntime[i];	storedRuntime[i] = "Blank";
-		tempProc[i] = storedProc[i];		storedProc[i] = "Blank";
-		tempSource[i] = storedSource[i];	storedSource[i] = "Blank";
-		tempUsr[i] = storedUsr[i];			storedUsr[i] = "Blank";
-		tempSrc[i] = storedSrc[i];			storedSrc[i] = "Blank";
-		tempNumRuntime[i] = numRuntime[i];	numRuntime[i] = 0;
-	}
-
-	while(highestCount > 0)
-	{
-		for(int i=0; i <= maxStorage; i++) //For every runtime
-		{
-			if(tempNumRuntime[i] == highestCount) //If the number of occurances of that runtime is equal to our current highest
-			{
-				for(int j=0; j <= maxStorage; j++) //Find the next available slot and store the info

```

Comment:
https://tgstation13.org/parsed-logs/sybil/runtimes/runtime-2016-04-28-condensed.txt
this file would have taken 1407321^15646^^2 iterations
for a better sense of scale, 15646^^2 is a number with about 65 thousand digits

that's a lot of iterations